examples/pna-example-exactmap-ternarymap.p4

```diff
@@ -0,0 +1,248 @@
+    inout pna_pre_output_metadata_t ostd)
+{
+    apply {
+        // No IPsec decryption for this example program, so pre
+        // control does nothing.
+    }
+}
+
+control MainControlImpl(
+    inout headers_t  hdr,
+    inout metadata_t meta,
+    in    pna_main_input_metadata_t  istd,
+    inout pna_main_output_metadata_t ostd)
+{
+
+    // Example instantiation from documentation comments for ExactMap
+    // extern in pna.p4
+    ExactMap<emap1_key, bit<16>>(
+        size = 1024,
+        const_entries = (list<exactmap_const_entry_t<emap1_key, bit<16>>>) {
+            {{ 5}, 10},  // key my_field= 5, value 10
+            {{ 6}, 27},  // key my_field= 6, value 27
+            {{10},  2}   // key my_field=10, value  2
+        },
+        default_value = 42)  // default value returned for all other keys
+    emap1;
+
+    ExactMap<emap2_key, emap2_val>(
+        size = 128,
+        initial_entries = (list<exactmap_initial_entry_t<emap2_key, emap2_val>>) {
+            // not const entry, key my_field=5, value {val1=10, val2=4}
+            {false, { 5}, {val1=10, val2=   4}},
+            // const entry, key my_field=6, value {val1=15, val2=4095}
+            { true, { 6}, {val1=15, val2=4095}},
+            // const entry, key my_field=10, value {val1=0, val2=28}
+            { true, {10}, {val1= 0, val2=  28}}
+        },
+        // default value returned for all other keys
+        default_value = {val1=9, val2=42})
+    emap2;
+
+    TernaryMap<tmap1_key, bit<16>>(
+        size = 1024,
+        const_entries = (list<ternarymap_const_entry_t<tmap1_key, bit<16>>>) {
+            // ternary key with
+            // + value 5, mask 0xffffff (exact match on least
+            //   significant 24 bits) for field f1,
+            // + value 2, mask 0x7 (exact match on all 3 bits)
+            //   for field f2,
+            // and value 10:
+            {{       5,   2},
+             {0xffffff, 0x7},
+             10},
+
+            // ternary key with
+            // + value 6, mask 0xff00ff (wildcard on middle 8
+            //   bits, exact match on the rest) for field f1,
+            // + value 2, mask 0x7 (exact match on all 3 bits)
+            //   for field f2,
+            // and value 27:
+            {{       6,   2},
+             {0xff00ff, 0x7},
+             27},
+            
+            // ternary key with
+            // + value 10, mask 0x00ffff (exact match on least
+            //   significant 16 bits, wildcard on bits
+            //   [23:16]),
+            // + value 0, mask 0x2 (exact match on only the
+            //   middle bit) for field f2,
+            // and value 2:
+            {{      10,   0},
+             {0x00ffff, 0x2},
+             2}
+        },
+        default_value = 42)  // default value returned for all other keys
+    tmap1;
+
+    TernaryMap<tmap2_key, tmap2_val>(
+        size = 1024,
+        largest_priority_wins = true,
+        initial_entries =
```

Comment:
This `initial_entries` parameter is intended to correspond with the new ability to define initial entries (without ` const` qualifier before the `entries` table property name) proposed in this PR for the language spec: https://github.com/p4lang/p4-spec/pull/1180

Here I do not see an easy way to make the per-entry `const` property or the per-entry `priority` optional, so they are required for each entry.